### DISCUSSION TITLE
Best way to preview a theme with all scopes

### DISCUSSION THREAD

**Discussion by ProfDoof** [helix-editor/helix#11799] [Q&A]
Hey y'all,

I have been working on designing a theme and wanted to see if anyone has any good resources to preview a theme and how it looks with every single scope.

One thing I had considered was writing a markdown file with a variety of different languages in different code blocks that would show off most if not all scopes. However, I'm not quite familiar enough with when those scopes are in play to be able to target them. 

Any other thoughts or suggestions?

**Comment by ProfDoof** [comment]
Also, kind of a side-note, any suggestions on how to test the accessibility of a theme? I want to at the bare minimum do a contrast comparison on the theme's values but it's not clear to me how I might approach that with something that lives in the terminal.

**Comment by rahil627** [comment]
i'm not sure what you mean by _scopes_.. i used [the syntax highlighting test files of bat](https://github.com/sharkdp/bat/tree/master/tests/syntax-tests/source), but maybe [learn x in y minutes](https://github.com/adambard/learnxinyminutes-docs) would've been more thorough.. also, it looks like they use markdown with code blocks as you planned! i looked into tree-sitter, but it was so complex... and.. _javascript_.. :sob:

**Reply by ProfDoof** [reply]
From my understanding, scopes are the names for the context that a style will be applied to. Scopes can have sub scopes. So, ui.background is a scope but so are all the different types and such.

**Reply by rahil627** [reply]
oooooh. i wish i looked this up earlier... here!

https://github.com/helix-editor/helix/blob/master/book/src/themes.md#scopes
